File: Algorithms/Implementation/Grading Students/GradingStudents.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    static int[] solve(int[] grades){
        // Complete this function
    	int n = grades.length;
    	int[] newScores = new int[n];
    	for(int i = 0; i<n; ++i){
    		int diff = 5 - grades[i]%5;
    		diff = (grades[i] >= 38 && diff < 3) ? diff : 0;
    		newScores[i] = grades[i] + diff;
    	}
    	return newScores;	
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] grades = new int[n];
        for(int grades_i=0; grades_i < n; grades_i++){
            grades[grades_i] = in.nextInt();
        }
        int[] result = solve(grades);
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + (i != result.length - 1 ? "\n" : ""));
        }
        System.out.println("");
    }
}

File: Algorithms/Implementation/Grading Students/GradingStudents.cpp
#include <bits/stdc++.h>

using namespace std;

vector < int > solve(vector < int > grades){
    // Complete this function
    int s = grades.size();
    vector<int> newScore(s);
    for(int i = 0; i < s; i++){
        int diff = 5 - grades[i]%5;
        diff  = (grades[i] >= 38 && (diff < 3)) ? diff : 0;
        newScore[i] = grades[i] + diff;
    }
    return newScore;
}

int main() {
    int n;
    cin >> n;
    vector<int> grades(n);
    for(int grades_i = 0; grades_i < n; grades_i++){
       cin >> grades[grades_i];
    }
    vector < int > result = solve(grades);
    for (ssize_t i = 0; i < result.size(); i++) {
        cout << result[i] << (i != result.size() - 1 ? "\n" : "");
    }
    cout << endl;
    return 0;
}
